### PR TITLE
Create WDB get command for packages and hotfixes

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#8555 #8570|

## Description
This PR implements the **get** command on WDB to obtain the agent's packages and hotfixes.
- The **packages get** command handles the option to request only the not triaged packages.
- The **packages get** command set every package as triaged at the end.
- Each command checks the sync_status of the table previous obtaining the information.
- Each command will respond the data using the new wdb_exec_stmt_send method to dispatch every row of information in a single command as soon as it is obtained.
- If there was a socket error sending the data, the socket is closed and no response is sent.
- Once the data is sent row by row, a final command is sent with the status of execution, including if the data was synced or not.
- Now **wdb** has a new parameter **peer** saved after blocking the database. This is the peer where the data will be sent row by row.
- Now, hotfixes and packages parsers were refactored.
- UT for wdb_agents.c and wdb_parser.c where added for every new code. 

## Tests

- Compilation without warnings in every supported platform
  - [X] Linux
- [X] Source installation
- [X] Package installation

- Memory tests for Linux
  - [X] Scan-build report
  - [X] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

**Comment by palaciosjeremias** [comment]
Valgrind result:
image

**Comment by palaciosjeremias** [comment]
Sacn-build report:
image

File: src/wazuh_db/wdb_parser.c [hunk 1]
```diff
@@ -3589,254 +3597,40 @@ int wdb_parse_ports(wdb_t * wdb, char * input, char * output) {
...
+    char* tail = NULL;
+    char* action = strtok_r(input, " ", &tail);
+
+    if (strcmp(action, "save") == 0) {
+        /* The format of the data is scan_id|scan_time|format|name|priority|section|size|vendor|install_time|version|architecture|multiarch|source|description|location*/
+        #define SAVE_PACKAGE_FIELDS_AMOUNT 16
+        char* fields[SAVE_PACKAGE_FIELDS_AMOUNT] = {NULL};
+        char* last = tail;
+
+        for (int i = 0; i < SAVE_PACKAGE_FIELDS_AMOUNT; i++) {
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_parser.c:3613]: Very nice improvement!

File: src/wazuh_db/wdb_parser.c [hunk 2]
```diff
@@ -3868,68 +3662,73 @@ int wdb_parse_packages(wdb_t * wdb, char * input, char * output) {
...
+            *output = '\0';
         }
 
-        scan_time = curr;
-        *next++ = '\0';
-        curr = next;
+        return result;
+    }
+    else {
+        mdebug1("Invalid Package info query syntax.");
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_parser.c:3698]: Just a minor thing. Change the capitalization of `Package` to `package`.

**Review by DProvinciani — APPROVED** [review]
Looks good!

File: src/wazuh_db/wdb.h [hunk 3]
```diff
@@ -864,8 +869,28 @@ int wdb_parse_osinfo(wdb_t * wdb, char * input, char * output);
 
 int wdb_parse_hardware(wdb_t * wdb, char * input, char * output);
 
+/**
+ * @brief Parses a packages command
+ * Commands:
+ * 1. delete: Deletes packages table
```
Comment (pereyra-m) [inline src/wazuh_db/wdb.h:875]: The parser in fact waits for `del`

File: src/wazuh_db/wdb.h [hunk 4]
```diff
@@ -864,8 +869,28 @@ int wdb_parse_osinfo(wdb_t * wdb, char * input, char * output);
...
+ * @param wdb Database of an agent
+ * @param input Buffer input
+ * @param output Buffer output
+ * */
 int wdb_parse_packages(wdb_t * wdb, char * input, char * output);
 
+/**
+ * @brief Parses a hotfixes command
+ * Commands:
+ * 1. delete: Deletes hotfixes table
```
Comment (pereyra-m) [inline src/wazuh_db/wdb.h:887]: Same comment about `del`

File: src/wazuh_db/wdb_parser.c (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_parser.c:3613]: +1

File: src/wazuh_db/wdb_parser.c [hunk 5]
```diff
@@ -3589,254 +3597,40 @@ int wdb_parse_ports(wdb_t * wdb, char * input, char * output) {
...
+
+        for (int i = 0; i < SAVE_PACKAGE_FIELDS_AMOUNT; i++) {
+            if (!(next = strtok_r(NULL, "|", &tail))) {
+                mdebug1("Invalid Package info query syntax.");
+                mdebug2("Package info query: %s", last);
+                snprintf(output, OS_MAXSTR + 1, "err Invalid Package info query syntax, near '%.32s'", last);
+                return result;
+            }
+            last = next;
+            if (strcmp(next, "NULL"))
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_parser.c:3621]: I'm not sure if this new implementation, replaces the string `"NULL"` with `NULL` for every field

**Review by pereyra-m — APPROVED** [review]
LGTM!

File: src/wazuh_db/wdb.h (hunk 3 above)
Comment (palaciosjeremias) [reply src/wazuh_db/wdb.h:875]: Fixed

File: src/wazuh_db/wdb.h (hunk 4 above)
Comment (palaciosjeremias) [reply src/wazuh_db/wdb.h:887]: Fixed

File: src/wazuh_db/wdb_parser.c (hunk 5 above)
Comment (palaciosjeremias) [reply src/wazuh_db/wdb_parser.c:3621]: In fact, this new implementation initializes every element of **fields** in NULL (line 3610), and line 3623 only assigns a different value if the string isn´t "NULL".
But has the same result.

File: src/wazuh_db/wdb_parser.c (hunk 2 above)
Comment (palaciosjeremias) [reply src/wazuh_db/wdb_parser.c]: Fixed

File: src/wazuh_db/wdb_parser.c (hunk 5 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_parser.c:3621]: You are right! I didn't read the `if (strcmp(next, "NULL"))` correctly.
Thank you!

**Pull request merged**